File: jtaf-web/src/java/ch/jtaf/model/Result.java
package ch.jtaf.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

@Entity
public class Result {

    @Id
    @GeneratedValue
    private Long id;
    private String result;
    private int points;
    @ManyToOne
    private Event event;
    @ManyToOne
    private Competition competition;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getResult() {
        return result;
    }

    public void setResult(String result) {
        this.result = result;
    }

    public int getPoints() {
        return points;
    }

    public void setPoints(int points) {
        this.points = points;
    }

    public Event getEvent() {
        return event;
    }

    public void setEvent(Event event) {
        this.event = event;
    }

    public Competition getCompetition() {
        return competition;
    }

    public void setCompetition(Competition competition) {
        this.competition = competition;
    }
}

File: jtaf2/src/main/kotlin/ch/jtaf/entity/Result.kt
package ch.jtaf.entity

import javax.persistence.*
import javax.persistence.GenerationType.IDENTITY

@Entity
data class Result(
        @Id @GeneratedValue(strategy = IDENTITY)
        var id: Long? = null,

        var result: String = "",
        var points: Int = 0,

        @ManyToOne
        var athlete: Athlete? = null,

        @ManyToOne
        var category: Category? = null,

        @ManyToOne
        var competition: Competition? = null,

        var position: Int = 0
)